src/Bicep.Core/Emit/ParametersFileTemplateWriter.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+
+namespace Bicep.Core.Emit
+{
+    public class ParametersFileTemplateWriter : TemplateWriter
+    {
+        public ParametersFileTemplateWriter(SemanticModel semanticModel, EmitterSettings settings): base(semanticModel, settings)
+        {
+            this.context = new EmitterContext(semanticModel, settings);
+            this.settings = settings;
+        }
+
+        private readonly EmitterContext context;
+        private readonly EmitterSettings settings;
+
+        public void Write(JsonTextWriter writer, string existingContent)
+        {
+            // Template is used for calcualting template hash, template jtoken is used for writing to file.
+            var templateJToken = GenerateTemplate();
+            if (templateJToken.SelectToken(GeneratorMetadataPath) is not JObject generatorObject)
+            {
+                throw new InvalidOperationException($"generated template doesn't contain a generator object at the path {GeneratorMetadataPath}");
+            }
+
+            if (!string.IsNullOrWhiteSpace(existingContent))
+            {
+                var fileJToken = JToken.Parse(existingContent);
```

Comment:
I think we should skip them for now to be consistent with the Deploy command. We can update that behavior with user feedback. For now @polatengin, would you mind skipping params with the `@secure` decorator?

Reply:
Agreed:  Please skip secure parameters for now.

Reply:
thanks for the information, resolving this conversation for now, when we have user feedback that says otherwise, we can create a follow-up PR to address this issue 👍